TourMate/TourMate/Frontend/Plan/Views/PlansViews/PlansDayView.swift

```diff
@@ -0,0 +1,214 @@
+                if let offset = planIdToOffset[prevPlanId] {
+                    prevEndX += offset.x
+                }
+                planIdToOffset[currentPlanId] = CGPoint(x: prevEndX, y: currentStartY)
+            } else {
+                planIdToOffset[currentPlanId] = CGPoint(x: 0, y: currentStartY)
+            }
+        }
+    }
+
+    func createPlanView(_ plan: Plan) -> some View {
+        switch plan.planType {
+        case .accommodation:
+            let accommodationViewModel = PlanViewModel<Accommodation>(
+                plan: plan as! Accommodation, trip: tripViewModel.trip)
+            return AnyView(AccommodationView(accommodationViewModel: accommodationViewModel))
+        case .activity:
+            let activityViewModel = PlanViewModel<Activity>(
+                plan: plan as! Activity, trip: tripViewModel.trip)
+            return AnyView(ActivityView(activityViewModel: activityViewModel))
+        case .restaurant:
+            let restaurantViewModel = PlanViewModel<Restaurant>(
+                plan: plan as! Restaurant, trip: tripViewModel.trip)
+            return AnyView(RestaurantView(restaurantViewModel: restaurantViewModel))
+        case .transport:
+            let transportViewModel = PlanViewModel<Transport>(
+                plan: plan as! Transport, trip: tripViewModel.trip)
+            return AnyView(TransportView(transportViewModel: transportViewModel))
+        case .flight:
+            let flightViewModel = PlanViewModel<Flight>(
+                plan: plan as! Flight, trip: tripViewModel.trip)
+            return AnyView(FlightView(flightViewModel: flightViewModel))
+        }
+    }
+
+    func createUpvoteView(_ plan: Plan) -> some View {
+        switch plan.planType {
+        case .accommodation:
+            let accommodationViewModel = PlanViewModel<Accommodation>(plan: plan as! Accommodation, trip: tripViewModel.trip)
+            return AnyView(UpvotePlanView(viewModel: accommodationViewModel, displayName: false))
+        case .activity:
+            let activityViewModel = PlanViewModel<Activity>(plan: plan as! Activity, trip: tripViewModel.trip)
+            return AnyView(UpvotePlanView(viewModel: activityViewModel, displayName: false))
+        case .restaurant:
+            let restaurantViewModel = PlanViewModel<Restaurant>(plan: plan as! Restaurant, trip: tripViewModel.trip)
+            return AnyView(UpvotePlanView(viewModel: restaurantViewModel, displayName: false))
+        case .transport:
+            let transportViewModel = PlanViewModel<Transport>(plan: plan as! Transport, trip: tripViewModel.trip)
+            return AnyView(UpvotePlanView(viewModel: transportViewModel, displayName: false))
+        case .flight:
+            let flightViewModel = PlanViewModel<Flight>(plan: plan as! Flight, trip: tripViewModel.trip)
+            return AnyView(UpvotePlanView(viewModel: flightViewModel, displayName: false))
+        }
+    }
```

Comment:
I think we should have some util class to create these views, as they are present in both PlansListView and this.

Reply:
That way it'd be easier to decouple it from the trip in a subsequent PR